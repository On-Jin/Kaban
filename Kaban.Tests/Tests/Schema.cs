using HotChocolate.Execution;
using Kaban.Tests.Setup;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace Kaban.Tests.Tests;

[Collection(nameof(WebAppCollectionFixture))]
public class Schema : IAsyncLifetime
{
    private readonly IRequestExecutor _executor;

    private readonly Func<Task> _resetDatabase;

    public Schema(WebAppFactory factory)
    {
        _resetDatabase = factory.ResetDatabaseAsync;
        var serviceScope = factory.Services.CreateScope();
        _executor = serviceScope.ServiceProvider.GetRequiredService<IRequestExecutorResolver>()
            .GetRequestExecutorAsync().Result;
    }

    public Task InitializeAsync() => Task.CompletedTask;

    public Task DisposeAsync() => _resetDatabase();

    [Fact]
    public void SchemaChangeTest()
    {
        _executor.Schema.Print().MatchSnapshot();
    }
}