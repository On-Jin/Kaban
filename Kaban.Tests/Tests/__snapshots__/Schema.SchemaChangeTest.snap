schema {
  query: Query
}

type Author {
  name: String!
  books: [Book!]!
}

type Book {
  title: String!
}

type Query {
  authors: [Author!]!
  book: Book!
}
